lib/matplotlib/image.py

```diff
@@ -1586,9 +1586,11 @@ def imsave(fname, arr, vmin=None, vmax=None, cmap=None, format=None,
             # because that's what backend_agg passes, and can be in fact used
             # as is, saving a few operations.
             rgba = arr
+        elif arr.ndim == 3 and arr.shape[-1] in (3,4):
```

Comment:
Does this work for `shape[-1] == 3`? AFAICS, rgba should have the alpha component included eventually?